### PR TITLE
Gitcoin

### PR DESCRIPTION
**Pull request by Jack-Works**
# Publish

- [x] Alice publishes the URL in an encrypted post, or publishes the URL directly on the SNS.
- [x] Alice sees the Gitcoin PAC under the post.

# Donate

- [x] Bob clicks the "Donate" button.
- [x] Maskbook opens the PSM of Gitcoin.
- [x] Bob selects a Wallet.
- [x] Bob inputs a value.
- [x] Bob clicks "Donate" button.
- [x] Maskbook presents Bob a Basic Wallet Transfer Request dialogue.
- [x] Bob clicks "Confirm" button.
- [x] The PSM of Gitcoin closes.
- [x] The PAC shows a success message and a share button.

# Share
- [x] Bob clicks the "Share" button in the PAC or the PSM.
- [x] Maskbook writes the URL (if the user ever donated, the text should be "I donated ${amount} ${token} for campaign "${title} on Gitcoin": ${url}", where the data comes from the most recent donation) of the Gitcoin campaign into the clipboard.
- [x] Sharing the URL in a post (encrypted or clear) will let other Maskbook users see the same PAC.

### CODE REVIEW

**Comment by neruthes** [comment]
> Maskbook writes the URL (if the user ever donated, the text should be "I donated ${amount} ${token} for campaign "${title} on Gitcoin": ${url}", where the data comes from **the most recent donation** of the Gitcoin campaign into the clipboard.

Should be "the most recent one among the donations which are made by local wallets".

File: src/setup.ui.ts [hunk 1]
```diff
@@ -4,6 +4,20 @@ import './provider.ui'
 import { LiveSelector, Watcher, DOMProxy } from '@holoflows/kit/es'
 import { exclusiveTasks } from './extension/content-script/tasks'
 
+// Jesus.png
```
Comment (guanbinrui) [inline src/setup.ui.ts:7]: This snippet should be labeled as 'FIXME'.

**Comment by neruthes** [comment]
Probably we want to improve the "view transaction history" feature in wallets. 🤔

**Comment by yisiliu** [comment]
To retrieve the metadata of a single grant, we can query the following apis in order:

1. `https://gitcoin.co/api/v0.1/grants/159/` to get `contract_address `, `description `, `title`, `amount_received `
2. `https://gitcoin.co/api/v0.1/grants/report/?eth_address=contract_address` to get `grantee/transactions` and compute the length of `transactions` ref:https://user-images.githubusercontent.com/36922376/81472705-7569ea00-9217-11ea-96a6-587744ac15b7.png

**Comment by neruthes** [comment]
## Note

I slightly modified the wording here.

"No one will know which wallet you used to donate"

"By using this service, you will also be contributing 5% of your contribution to the Gitcoin Grants Round 6+ Development Fund"

image

**Comment by Tedko** [comment]
> ## Note
> I slightly modified the wording here.
> 
> "No one will know which wallet you used to donate"
> 
> "By using this service, you will also be contributing 5% of your contribution to the Gitcoin Grants Round 6+ Development Fund"
> 
> image

Screen Shot 2020-05-18 at 17 17 47
Let's just use wording from gitcoin instead of modify these

**Comment by Tedko** [comment]
> ## Note
> I slightly modified the wording here.
> 
> "No one will know which wallet you used to donate"
> 
> "By using this service, you will also be contributing 5% of your contribution to the Gitcoin Grants Round 6+ Development Fund"
> 
> image

It (the wallet addr and the tx) will be still on chain. I guess here the major difference is on the UX(front end) of gitcoin and the API, gitcoin will not record the addr for privacy.

**Comment by neruthes** [comment]
> It (the wallet addr and the tx) will be still on chain. I guess here the major difference is on the UX(front end) of gitcoin and the API, gitcoin will not record the addr for privacy.

Good point. We need to figure out the truth about this before deciding the wording of the option.

**Comment by Tedko** [comment]
take a look of a Gitcoin grant - transactions. You will see some tx with
wallet addr (but you can just use the recipient addr to find the sender)

**Comment by Jack-Works** [comment]
@guanbinrui rebased

**Comment by neruthes** [comment]
image

Maybe @Jack-Works wants to mark more items as completed? 🤔

**Comment by guanbinrui** [comment]
@Jack-Works Rebased onto the master branch.

**Comment by neruthes** [comment]
Will we be able to produce build for internal test today?

**Comment by Jack-Works** [comment]
@guanbinrui does it able to use?

Before we release a version for GitCoin, must wait for #1200 to be merged

**Comment by guanbinrui** [comment]
> @guanbinrui does it able to use?
> 
> Before we release a version for GitCoin, must wait for #1200 to be merged

It's ready now.

**Pull request merged**